### ISSUE TITLE
Support .hlint.yaml in nested folders

### ISSUE THREAD

**Issue by fredericcogny** [ndmitchell/hlint#594]
Hi Neil

In the case where we have a big repo with subfolders containing different packages I would like to be able to customize/override the top level `.hlint.yaml` file at the level of each package, yet,  run `hlint` only once at the top level rather than once per subfolder. This would make the implementation ni the CI notably simpler

![image](https://user-images.githubusercontent.com/15909740/52494083-4d7f7700-2bcd-11e9-86fd-ecd4f1d2a9bd.png)

**Comment by ndmitchell** [comment]
Thanks Fred, sounds like a good idea. Unfortunately, because of the way the .hlint.yaml file is currently handled (before any files are looked at) it's currently very difficult to do that. But I plan to revise that whole area at some point, and adding this feature sounds sensible.

My inclination is that if you have a .hlint.yaml in mathjs, and another at the root, it should combine both when processing mathjs/src or similar.

**Comment by fredericcogny** [comment]
I agree. - not sure what internal representation you have but morally it should be the union of the yaml with a bias towards the nested files in case of common keys.

**Comment by ndmitchell** [comment]
That's pretty much I was imagining.
